source/extensions/stat_sinks/common/statsd/statsd.cc

```diff
@@ -20,22 +20,22 @@ namespace StatSinks {
 namespace Common {
 namespace Statsd {
 
-Writer::Writer(Network::Address::InstanceConstSharedPtr address) {
-  fd_ = address->socket(Network::Address::SocketType::Datagram);
-  ASSERT(fd_ != -1);
+Writer::Writer(Network::Address::InstanceConstSharedPtr address)
+    : io_handle_(address->socket(Network::Address::SocketType::Datagram)) {
+  ASSERT(io_handle_->fd() != -1);
 
-  const Api::SysCallIntResult result = address->connect(fd_);
+  const Api::SysCallIntResult result = address->connect(io_handle_->fd());
   ASSERT(result.rc_ != -1);
 }
 
 Writer::~Writer() {
-  if (fd_ != -1) {
-    RELEASE_ASSERT(close(fd_) == 0, "");
+  if (io_handle_->fd() != -1) {
+    RELEASE_ASSERT(close(io_handle_->fd()) == 0, "");
```

Comment:
When you assert the handle has fd_ == -1, you will need to close it here also.

Reply:
Good point!